src/server/service/kv.rs

```diff
@@ -160,6 +160,12 @@ impl<T: RaftStoreRouter<E::Local> + 'static, E: Engine, L: LockManager> Service<
         match ch.send_raft_msg(msg) {
             Ok(()) => Ok(()),
             Err(RaftStoreError::RegionNotFound(_)) => Ok(()),
+            Err(RaftStoreError::Transport(DiscardReason::Full)) => {
+                CHANNEL_FULL_COUNTER_VEC
+                    .with_label_values(&[&store_id.to_string()])
+                    .inc();
```

Comment:
Seems this metric can be updated in `handle_send_error` in `router.rs` in order to detect the channel full errors in `send_raft_msg` and `send_command` from all callers.

Reply:
Ok any labels to be added to provide info about the source of channel full errors if we are detecting from all_callers?

Also what would be a good replacement for the existing label value (we used channel name [earlier](https://github.com/tikv/tikv/pull/4232/files#diff-2d7955eb02fa5282d80c7917a02c45bb5a669a7e5fea181d176e38e664297723L122) which is no longer available).

Reply:
We don't care about the `store_id` because it's always the local id. But we care about which type of channel is full, namely, normal channel or control channel.
You can the code(it's better to add the metric here)https://github.com/tikv/tikv/blob/master/components/raftstore/src/store/fsm/store.rs#L226-L285.

Reply:
I've updated the metrics in `components/batch-system/src/router.rs` which is being used by the code you referenced.
Using `control` & `normal` as the values of the type label.